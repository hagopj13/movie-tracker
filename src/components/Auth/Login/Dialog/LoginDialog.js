import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { makeStyles } from '@material-ui/core/styles';
import DialogContent from '@material-ui/core/DialogContent';
import Typography from '@material-ui/core/Typography';
import Link from '@material-ui/core/Link';

import Dialog from '../../../UI/Dialog/Dialog';
import LoginForm from './Form/LoginForm';
import AuthActionTypes from '../../../../store/auth/auth.types';
import { login, loginClear } from '../../../../store/auth/auth.actions';
import { createIsLoadingSelector } from '../../../../store/api/loading/loading.selectors';
import { createErrorSelector } from '../../../../store/api/error/error.selectors';

type Props = {
  isOpen: boolean,
  isLoginLoading: boolean,
  loginError: string,
  onClose: () => void,
  onLogin: ({ username: string, password: string }) => void,
  onLoginClear: () => void,
};

const useStyles = makeStyles((theme) => ({
  content: {
    padding: theme.spacing(0.5, 5, 1),
    marginBottom: theme.spacing(2),
  },
  contentText: {
    marginBottom: theme.spacing(2),
    maxWidth: '45ch',
  },
}));

const DIALOG_TITLE = 'Login';

const LoginDialog = (props: Props) => {
  const { isOpen, isLoginLoading, loginError, onClose, onLogin, onLoginClear } = props;
  const classes = useStyles();

  const renderInfoText = () => (
    <Typography className={classes.contentText}>
      This app gets its data from the TMDb APIs. To view your account information, login with your
      TMDb credentials in the form below. To create one,&nbsp;
      <Link target="_blank" href="https://www.themoviedb.org/account/signup" color="primary">
        click here
      </Link>
      .
    </Typography>
  );

  return (
    <Dialog isOpen={isOpen} onClose={onClose} title={DIALOG_TITLE} onExited={onLoginClear}>
      <DialogContent className={classes.content}>
        {renderInfoText()}
        <LoginForm isLoginLoading={isLoginLoading} loginError={loginError} onSubmit={onLogin} />
      </DialogContent>
    </Dialog>
  );
};

const mapStateToProps = createStructuredSelector({
  isLoginLoading: createIsLoadingSelector([AuthActionTypes.LOGIN]),
  loginError: createErrorSelector([AuthActionTypes.LOGIN]),
});

const mapDispatchToProps = {
  onLogin: login,
  onLoginClear: loginClear,
};

export default connect(mapStateToProps, mapDispatchToProps)(LoginDialog);
