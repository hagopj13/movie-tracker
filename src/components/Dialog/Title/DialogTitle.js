// @flow
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import MuiDialogTitle from '@material-ui/core/DialogTitle';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';

type Props = {
  title: string,
  onClose: () => void,
};

const useStyles = makeStyles((theme) => ({
  closeButton: {
    position: 'absolute',
    right: theme.spacing(1),
    top: theme.spacing(1),
    color: theme.palette.grey[500],
  },
}));

const DialogTitle = (props: Props) => {
  const { title, onClose, ...otherProps } = props;
  const classes = useStyles();

  return (
    <MuiDialogTitle disableTypography {...(otherProps: any)}>
      <Typography variant="h6">{title}</Typography>
      <IconButton aria-label="close" className={classes.closeButton} onClick={onClose}>
        <CloseIcon />
      </IconButton>
    </MuiDialogTitle>
  );
};

export default DialogTitle;
