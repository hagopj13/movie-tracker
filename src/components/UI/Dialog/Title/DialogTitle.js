import React from 'react';
import type { Node } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import MuiDialogTitle from '@material-ui/core/DialogTitle';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';

type Props = {
  children: Node,
  onClose: () => void,
};

const useStyles = makeStyles((theme) => ({
  closeButton: {
    position: 'absolute',
    right: theme.spacing(1),
    top: theme.spacing(1),
    color: theme.palette.grey[500],
  },
}));

const DialogTitle = (props: Props) => {
  const { children, onClose, ...otherProps } = props;
  const classes = useStyles();

  return (
    <MuiDialogTitle disableTypography {...otherProps}>
      <Typography variant="h6">{children}</Typography>
      {onClose ? (
        <IconButton aria-label="close" className={classes.closeButton} onClick={onClose}>
          <CloseIcon />
        </IconButton>
      ) : null}
    </MuiDialogTitle>
  );
};

export default DialogTitle;
