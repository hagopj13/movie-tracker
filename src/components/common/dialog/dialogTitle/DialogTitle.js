import React from 'react';
import type { Node } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import MuiDialogTitle from '@material-ui/core/DialogTitle';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';

type Props = {
  children: Node,
  onClose: () => void,
};

const useStyles = makeStyles((theme) => ({
  root: {
    margin: 0,
    padding: theme.spacing(2),
  },
  closeButton: {
    position: 'absolute',
    right: theme.spacing(1),
    top: theme.spacing(1),
    color: theme.palette.grey[500],
  },
}));

const DialogTitle = (props: Props) => {
  const { children, onClose, ...otherProps } = props;
  const classes = useStyles();

  return (
    <MuiDialogTitle disableTypography className={classes.root} {...otherProps}>
      <Typography variant="h6">{children}</Typography>
      {onClose ? (
        <IconButton aria-label="close" className={classes.closeButton} onClick={onClose}>
          <CloseIcon />
        </IconButton>
      ) : null}
    </MuiDialogTitle>
  );
};

export default DialogTitle;
