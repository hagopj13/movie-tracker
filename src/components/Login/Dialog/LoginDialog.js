// @flow
import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { makeStyles } from '@material-ui/core/styles';
import DialogContent from '@material-ui/core/DialogContent';
import Typography from '@material-ui/core/Typography';
import Link from '@material-ui/core/Link';

import Dialog from 'components/Dialog/Dialog';
import AuthActionTypes from 'store/auth/auth.types';
import authActions from 'store/auth/auth.actions';
import loadingSelectors from 'store/api/loading/loading.selectors';
import errorSelectors from 'store/api/error/error.selectors';

import LoginForm from '../Form/LoginForm';

type Props = {
  isOpen: boolean,
  isLoginLoading: boolean,
  loginError: string,
  onClose: () => void,
  onLogin: ({ username: string, password: string }) => void,
  onLoginClear: () => void,
};

const useStyles = makeStyles((theme) => ({
  content: {
    padding: theme.spacing(0.5, 5, 1),
    marginBottom: theme.spacing(2),
  },
  contentText: {
    marginBottom: theme.spacing(2),
    maxWidth: '45ch',
  },
}));

const DIALOG_TITLE = 'Login';

export const LoginDialog = (props: Props) => {
  const { isOpen, isLoginLoading, loginError, onClose, onLogin, onLoginClear } = props;
  const classes = useStyles();

  const renderInfoText = () => (
    <Typography className={classes.contentText}>
      This app gets its data from the TMDb APIs. To view your account information, login with your
      TMDb credentials in the form below. To create one,&nbsp;
      <Link target="_blank" href="https://www.themoviedb.org/account/signup" color="primary">
        click here
      </Link>
      .
    </Typography>
  );

  return (
    <Dialog isOpen={isOpen} onClose={onClose} title={DIALOG_TITLE} onExited={onLoginClear}>
      <DialogContent className={classes.content}>
        {renderInfoText()}
        <LoginForm isLoginLoading={isLoginLoading} loginError={loginError} onSubmit={onLogin} />
      </DialogContent>
    </Dialog>
  );
};

const mapStateToProps = createStructuredSelector({
  isLoginLoading: loadingSelectors.createIsLoadingSelector([AuthActionTypes.LOGIN]),
  loginError: errorSelectors.createErrorSelector([AuthActionTypes.LOGIN]),
});

const mapDispatchToProps = {
  onLogin: authActions.login,
  onLoginClear: authActions.loginClear,
};

export default connect(mapStateToProps, mapDispatchToProps)(LoginDialog);
