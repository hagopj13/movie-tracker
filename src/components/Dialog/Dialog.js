// @flow
import React from 'react';
import type { Node } from 'react';
import MuiDialog from '@material-ui/core/Dialog';

import DialogTitle from './Title/DialogTitle';

type Props = {
  isOpen: boolean,
  title: string,
  children: Node,
  onClose: () => void,
  ...
};

const Dialog = (props: Props) => {
  const { isOpen, title, onClose, children, ...dialogProps } = props;

  return (
    <MuiDialog open={isOpen} onClose={onClose} disableRestoreFocus {...(dialogProps: any)}>
      <DialogTitle title={title} onClose={onClose} />
      {children}
    </MuiDialog>
  );
};

export default Dialog;
