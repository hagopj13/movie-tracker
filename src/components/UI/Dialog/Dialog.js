import React from 'react';
import MuiDialog from '@material-ui/core/Dialog';

import DialogTitle from './Title/DialogTitle';

type Props = {
  isOpen: boolean,
  title: string,
  children: React.ReactNode,
  onClose: () => void,
};

const Dialog = (props: Props) => {
  const { isOpen, title, onClose, children, ...dialogProps } = props;

  return (
    <MuiDialog open={isOpen} onClose={onClose} disableRestoreFocus {...dialogProps}>
      <DialogTitle onClose={onClose}>{title}</DialogTitle>
      {children}
    </MuiDialog>
  );
};

export default Dialog;
