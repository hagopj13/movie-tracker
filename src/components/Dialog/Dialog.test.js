import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import MuiDialog from '@material-ui/core/Dialog';

import Dialog from './Dialog';
import DialogTitle from './Title/DialogTitle';

describe('Dialog component', () => {
  let shallow;
  let isOpen;
  let title;
  let wrapper;

  const FakeDialogContent = () => <div>Dialog content</div>;
  const mockClose = jest.fn();

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    jest.clearAllMocks();
    isOpen = true;
    title = 'Some title';
    const mockProps = {
      isOpen,
      title,
      onClose: mockClose,
    };
    wrapper = shallow(
      <Dialog {...mockProps}>
        <FakeDialogContent />
      </Dialog>,
    );
  });

  it('should render the Dialog component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the child component', () => {
    expect(wrapper.exists(FakeDialogContent)).toBe(true);
  });

  it('should pass isOpen as a prop to MuiDialog', () => {
    expect(wrapper.find(MuiDialog).prop('open')).toBe(isOpen);
  });

  it('should pass title as a prop to DialogTitle', () => {
    expect(wrapper.find(DialogTitle).prop('title')).toBe(title);
  });

  it('should call onClose when onClose of MuiDialog is triggered', () => {
    wrapper.find(MuiDialog).prop('onClose')();
    expect(mockClose).toHaveBeenCalledTimes(1);
  });
});
