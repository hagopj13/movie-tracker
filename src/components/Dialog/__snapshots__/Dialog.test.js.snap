// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog component should render the Dialog component correctly 1`] = `
<WithStyles(ForwardRef(Dialog))
  disableRestoreFocus={true}
  onClose={[MockFunction]}
  open={true}
>
  <DialogTitle
    onClose={[MockFunction]}
    title="Some title"
  />
  <FakeDialogContent />
</WithStyles(ForwardRef(Dialog))>
`;
